File: front/reserve_app/src/domain/entities/Role.js
export class Role {
    constructor(data) {
        if (!data) {
            throw new Error('Role data is required');
        }

        this.id = data.id;
        this.name = data.name;
        this.code = data.code;
        this.description = data.description;
        this.level = data.level;
        this.isSystem = data.is_system;
        this.scopeId = data.scope_id;
        this.scopeName = data.scope_name;
        this.scopeCode = data.scope_code;
    }

    isHighLevel() {
        return this.level >= 90;
    }

    toString() {
        return this.name;
    }
} 

File: mobile/rupu/lib/domain/entities/role.dart
class Role {
  final int id;
  final String name;
  final String code;
  final String description;
  final int level;
  final int? scopeId;
  final String? scopeName;
  final String? scopeCode;
  final bool isSystem;
  final int? businessTypeId;
  final String? businessTypeName;

  const Role({
    required this.id,
    required this.name,
    required this.code,
    required this.description,
    required this.level,
    this.scopeId,
    this.scopeName,
    this.scopeCode,
    this.isSystem = false,
    this.businessTypeId,
    this.businessTypeName,
  });
}

class RolesCatalog {
  final List<Role> roles;
  final int count;

  const RolesCatalog({
    required this.roles,
    required this.count,
  });
}
